extension/parquet/zstd_file_system.cpp

```diff
@@ -28,7 +28,15 @@ ZstdStreamWrapper::~ZstdStreamWrapper() {
 	}
 	try {
 		Close();
-	} catch (...) { // NOLINT: swallow exceptions in destructor
+	} catch (std::exception &ex) {
+		if (file && file->child_handle) {
+			// FIXME: make more log context available here.
+			ErrorData data(ex);
+			DUCKDB_LOG_ERROR(file->child_handle->logger,
+			                 "ZstdStreamWrapper::~ZstdStreamWrapper()\t\t" + data.Message());
+		}
```

Comment:
Just a note: is there any guarantee that DUCKDB_LOG_ERROR should not throw (=noexcept)? I don't think at the moment, but maybe it should.

I'd say either `DUCKDB_LOG_ERROR` has a `try {} catch (...)` or we need one explicitly here (nested ones?).

Basically I'd say throwing within a destructor is at least unexpected, and possibly a disaster coming.